File: app/src/main/java/at/droelf/travellogapp/ImageUploadAndroidService.java
package at.droelf.travellogapp;

import android.app.IntentService;
import android.content.Intent;

import org.springframework.web.client.RestClientException;

public class ImageUploadAndroidService extends IntentService {

    private final ImageUploadService imageUploadService = ImageUploadService.getInstance();
    private final NetworkClient networkClient = new NetworkClient();

    public ImageUploadAndroidService() {
        super("ImageUploadAndroidService");
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        for (UploadImage uploadImage : imageUploadService.getQueuedImages()) {
            try {
                boolean successfulUpload = networkClient.uploadImage(uploadImage.dateTime, uploadImage.timezone, uploadImage.name, uploadImage.imagePath);

                if(successfulUpload) {
                    imageUploadService.setImageUploaded(uploadImage);
                }
            }
            catch (RestClientException e) {
                e.printStackTrace();
            }
        }
        // TODO read from disk and upload image and mark as uploaded
    }
}

File: app/src/main/scala/at/droelf/travellogapp/ImageUploadAndroidService.scala
package at.droelf.travellogapp

import android.app.IntentService
import android.content.Intent
import org.springframework.http.HttpStatus
import org.springframework.web.client.RestClientException

import scala.util.{Failure, Success, Try}

class ImageUploadAndroidService extends IntentService("ImageUploadAndroidService") {

  val imageUploadService: ImageUploadService = ImageUploadService.getInstance
  val networkClient: NetworkClient = new NetworkClient

  protected def onHandleIntent(intent: Intent) {
    imageUploadService.getQueuedImages.map( queuedImage => {

      Try[HttpStatus] {
        networkClient.uploadImage(queuedImage.dateTime, queuedImage.timezone, queuedImage.name, queuedImage.imagePath)
      } match {
        case Success(httpStatus) => if(httpStatus == HttpStatus.OK) imageUploadService.setImageUploaded(queuedImage.id)
        case Failure(e) => e.printStackTrace()
      }

    })
  }

}